### PR TITLE
generate ts bindings for interfaces in npm package

### PR DESCRIPTION
**Pull request by RnkSngh**
PR to add interface bindings to our npm package - the use case for this was in the [fee design api](https://github.com/polymerdao/fee-estimator/pull/2#discussion_r1640166923)

The only manual changes were in the package.json, but this PR includes the additional contract changes

### CODE REVIEW

**Review by dshiell — APPROVED** [review]
lgtm!

File: package.json [hunk 1]
```diff
@@ -34,7 +34,7 @@
     "lint": "solhint contracts/**/*.sol",
     "test": "forge test",
     "build": "npm run build-ts-contract-bindings && npm run build-go-contract-bindings && tsup",
-    "build-ts-contract-bindings": "npm run build-contracts && typechain --target ethers-v6 --out-dir src/evm/contracts/ './out/?(OptimisticProofVerifier|ProofVerifier|Ibc|IbcUtils|Channel|Dispatcher|Mars|Earth|UniversalChannelHandler|DummyProofVerifier|DummyLightClient|ERC1967Proxy|OptimisticLightClient).sol/*.json'",
+    "build-ts-contract-bindings": "npm run build-contracts && typechain --target ethers-v6 --out-dir src/evm/contracts/ './out/?(OptimisticProofVerifier|ProofVerifier|Ibc|IbcUtils|Channel|Dispatcher|Mars|Earth|UniversalChannelHandler|DummyProofVerifier|DummyLightClient|ERC1967Proxy|OptimisticLightClient|IFeeVault|IbcDispatcher|IDispatcher|ILightClient|IProofVerifier|IUniversalChannelHandler).sol/*.json'",
     "build-go-contract-bindings": "echo go bindings generation not yet implemented!",
```
Comment (h5law) [inline package.json:38]: You should rebase onto main to have go bindings generate too.

**Review by h5law — COMMENTED** [review]
LGTM just need to rebase for latest workflows etc 👍🏼

File: package.json (hunk 1 above)
Comment (RnkSngh) [reply package.json:38]: ah yes, good idea

**Pull request merged**